pkg/endpoint/endpoint.go

```diff
@@ -694,10 +694,6 @@ func (e *Endpoint) GetLabels() labels.Labels {
 // GetSecurityIdentity returns the security identity of the endpoint. It assumes
 // the endpoint's mutex is held.
 func (e *Endpoint) GetSecurityIdentity() (*identity.Identity, error) {
-	if err := e.rlockAlive(); err != nil {
```

Comment:
Assuming we resolve the above, it seems like we may also want to revisit the code comment above this function. It already looks potentially out-of-date but if this PR changes the locking behavior then the comment should be fixed to match.